res/controllers/Traktor-Kontrol-S2-MK3-hid-scripts.js

```diff
@@ -670,19 +670,92 @@ TraktorS2MK3.wheelDeltas = function(deckNumber, value) {
 };
 
 TraktorS2MK3.fxHandler = function(field) {
+    /* We support 8 effects in total by having 2 effects per fx button. *
+     * First press of the button will load the preset at the index of the quick effect preset list *
+     * Second press will load the preset index + 4 *
+     * Arrange your quick effect preset list from 1 to 8 like this: 1/5, 2/6, 3/7, 4/8 */
     if (field.value === 0) {
         return;
     }
 
+    const availableGroups = ["[Channel1]", "[Channel2]"];
+    const fxButtonCount = 4;
+    let targetGroups = [];
     const fxNumber = parseInt(field.id[field.id.length - 1]);
-    const group = "[EffectRack1_EffectUnit" + fxNumber + "]";
+    let noShift = false;
 
-    // Toggle effect unit
-    TraktorS2MK3.fxButtonState[fxNumber] = !TraktorS2MK3.fxButtonState[fxNumber];
+    // Target decks whose shift button is pressed.
+    for (const group of availableGroups) {
+        if (TraktorS2MK3.shiftPressed[group]) { targetGroups.push(group); }
+    }
+
+    // If no shift button was pressed fall back to target all decks.
+    if (targetGroups.length === 0) {
+        targetGroups = availableGroups;
+        noShift = true;
+    }
```

Comment:
Wouldn't it make sense to check here that the effect knob is in zero position (or close to it)? Otherwise we could reassign the effect chain preset, while it's audible.

Reply:
IMO this is up to the user to decide, and the learing curve should be pretty steep ;)
Related: #11198

Reply:
I'm not sure how this works currently, but when I used the code from my old PR knobs would always reset when switching effect, whether that be to the center position or the zero position and the knob would only control it again when moved back to the initial point.